src/components/radioButton/RadioGroup.tsx

```diff
@@ -35,10 +42,24 @@ class RadioGroup extends PureComponent<Props, RadioGroupState> {
     };
   }
 
-  UNSAFE_componentWillReceiveProps(nextProps: Props) {
-    if (this.props.initialValue !== nextProps.initialValue) {
-      this.setState({value: nextProps.initialValue});
+  static getUpdatedState = (
+    nextProps: Props,
+    prevState: RadioGroupState
+  ): RadioGroupState | null => {
+    const {value} = prevState;
+    const {initialValue} = nextProps;
+    
+    if (_.isUndefined(nextProps.initialValue) || value === initialValue) {
```

Comment:
Why no turn the component to uncontrolled component? Where the `initialValue` is literally **initial** value and not updated through a prop change?

Reply:
@Inbal-Tish, Yeah, I though about it, but then we going to lose the use case to let users 'reset' the component value. 
I consult with Ethan to have a second opinion and we said that this use-cases should be kept